mne/io/pick.py

```diff
@@ -829,7 +829,7 @@ def _picks_to_idx(info, picks, none='data', exclude='bads', allow_empty=False,
         raise ValueError('picks must be 1D, got %sD' % (picks.ndim,))
     if picks.dtype.char in ('S', 'U'):
         picks = _picks_str_to_idx(info, picks, exclude, with_ref_meg)
-    if picks.dtype != np.int:
+    if picks.dtype.kind not in ['i', 'u']:
```

Comment:
I would add after this conditional an explicit cast:
```
picks = picks.astype(int)
```
There might be places where we reasonably assume `picks` is `int` dtype.

Reply:
I included it!